subt/cloudsim2osgar.py

```diff
@@ -58,21 +60,45 @@ def publish(self, channel, data):
 
 
 class main:
-    def __init__(self):
+    def __init__(self, robot_name):
         # get cloudsim ready
-        robot_name = sys.argv[1]
-        imu_name = '/'+robot_name+'/imu/data'
         wait_for_master()
-        rospy.init_node('imu2osgar', log_level=rospy.DEBUG)
-        rospy.loginfo("waiting for {}".format(imu_name))
-        rospy.wait_for_message(imu_name, Imu)
-        rospy.sleep(2)
-        self.imu_count = 0
-
-        # start
+        rospy.init_node('cloudsim2osgar', log_level=rospy.DEBUG)
         self.bus = Bus()
-        self.bus.register('rot', 'acc', 'orientation')
-        rospy.Subscriber(imu_name, Imu, self.imu)
+
+        # common topics
+        topics = [
+            ('/' + robot_name + '/imu/data', Imu, self.imu, ('rot', 'acc', 'orientation')),
+        ]
+
+        # configuration specific topics
+        robot_description = rospy.get_param("/{}/robot_description".format(robot_name))
+        if "robotika_x2_sensor_config_1" in robot_description:
+            rospy.loginfo("robotika x2")
+        elif "ssci_x2_sensor_config_1" in robot_description:
+            rospy.loginfo("ssci x2")
+        elif "ssci_x4_sensor_config_2" in robot_description:
+            rospy.loginfo("ssci drone")
+            topics.append(('/' + robot_name + '/top_scan', LaserScan, self.top_scan, ('top_scan',)))
+            topics.append(('/' + robot_name + '/bottom_scan', LaserScan, self.bottom_scan, ('bottom_scan',)))
+        elif "TeamBase" in robot_description:
+            rospy.loginfo("teambase")
+        elif "robotika_freyja_sensor_config_1" in robot_description:
```

Comment:
can you please add also Freyja 2 (robot with breadcrumbs), which I am testing at the moment? It has now the same inputs ... but can be postponed ...